designing page prep for expressive sunset

janchild opened the pull request:
This PR updates the `Designing` page to ready the page for changes coming with the Expressive Theme sunset.

janchild commented:
I've tightened up this page to create a narrative flow for a designer coming in and getting set up the first time, or coming back to find new information. 

There is [another ticket](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6445) detailing future changes to the Designing page, possibly breaking the content into separate pages but this would be better placed after we have some new steps to take for kits and new resources.

shixiedesign commented:
Capturing the notes from the other ticket here and closing that one for now: 

![image](https://user-images.githubusercontent.com/44503588/124025481-1a182f80-d9b6-11eb-8db0-7a4db7fcb0b7.png)

shixiedesign commented:
0. This is a fix that doesn't directly relate to expressive sunset, but if we can add a mention that our current components are hosted in **boxfolder**, click here to request for access, that would be great to patch the temporary unavailability of our kit

In general, I understand if we wanna avoid large operations right now, but it feels like "expressive kit" is too front and center. A few potential things to consider:

1. I think we might need a short explanation on what the "sketch kit" users are getting here. The mention of "Carbon expressive" is too easy to miss. Can we quickly explain: **the design kit of our component library is coming soon. For now, you can build our components with elements from the Carbon expressive kit.**

2. Should we repeat Expressive kit is going away above this section? Perhaps as a notification? 

![image](https://user-images.githubusercontent.com/15144993/131905147-84757073-b064-48f2-adfa-f132840b1042.png)

3. Lastly, I notice the "Learn about expressive styling" section. I wonder if this section should be the first one on top, prelude to "installing kit".

oliviaflory commented:
A minor suggestion: could we swap the position of the `IBM icons (24, 32px)` card and the `IBM icons (16, 20px)` card so the smaller sizes (16, 20) are first? The smaller size library is likely what teams will use the most.

<img width="728" alt="Screen Shot 2021-09-02 at 12 40 44 PM" src="https://user-images.githubusercontent.com/45692622/131906509-aac3dc44-a820-42d2-8c7d-7aed538379d6.png">

---

If we add the bit about other dotcom components being hosted in Box we might want to clarify that only IBMers will be able to access those folders. Happy to help figure out instructions/wording for that piece!

janchild commented:
@shixiedesign and @oliviaflory 

Thanks so much for the feedback! The next version is building and will be ready for the next round of reviews shortly. 
👁️👁️  

- Moved expressive styling section to the top
- Added some explanations of the Sketch kit and what is coming soon
- Added new H2 section `Get Carbon for IBM.components` and have added pointers to Sketch folders on Box. 
- Flipped the icon cards

shixiedesign commented:
Love the updates! Only 1 more comment on `Get Carbon for IBM.com components` section. We also provide "Design only" and "feature flag" components. I wonder if we should just share the entire "Components" folder here, perhaps as a large resource card? Alternatively we should at least add the "Design only" ones - they are the ones we hope adopters can pick up and build, or request us to build it at some point. 

![image](https://user-images.githubusercontent.com/15144993/131989170-1873d94c-4e05-4e82-8206-1339842e8906.png)

janchild commented:
> Love the updates! Only 1 more comment on `Get Carbon for IBM.com components` section. We also provide "Design only" and "feature flag" components. I wonder if we should just share the entire "Components" https://ibm.box.com/s/vyt75tjqx7mk6g7e8h3vw013zstcdk4s folder here, perhaps as a large resource card?
> 

Hey @shixiedesign, directing people to that top level folder might be overwhelming, especially for newbies. There are a lot of folders in there. 

I've been playing with mini resource cards for this section and this is looking good, and better than links.

<img width="757" alt="image" src="https://user-images.githubusercontent.com/44503588/132023131-d1cf936c-e3e1-4c39-870a-402f7e87297e.png">

Not sure which order we would want, presumably not alphabetical because I think the `Design only components` and the `Feature flag components` would be accessed the least. Anyway, will leave it there for further comment.

oliviaflory commented:
Agree that the Component folder might be overwhelming for newer users, or just might not be clear which folder they should look into first. We could write some descriptions in this section for what users will find inside each folder? 

**UI components**
Components that are designed and built to support or enable small UI tasks and actions. They can be paired with other components to create layout components, templates, and patterns.
> Swiped from the Component Overview page

**Layout components**
Layout components show some of the common layouts used in IBM’s digital space. They are carefully designed and coded to communicate prescribed content types and are expressions of IBM’s distinct brand ethos.
> Swiped from the Component Overview page

**Feature flag**
Components that are built under the feature flag branch and are being tested before adding to the library permanently. Warning: the Carbon for IBM.com team reserves the right to sunset components within the feature branch at any time.

**Design only**
Components that have a design, but are not available in code. We are always looking for adopters to contribute to the library, if your team would like to build any of these components to use in your experience, please reach out to the Carbon for IBM.com team to let us know you'll be contributing.

> _Just took a quick look at the Design only folder, I’m not sure every component is something we’d want contributed? Specifically Locale change banner?_

janchild commented:
> Agree that the Component folder might be overwhelming for newer users, or just might not be clear which folder they should look into first. We could write some descriptions in this section for what users will find inside each folder?

I can put these in and we can take a look.

> 
> > _Just took a quick look at the Design only folder, I’m not sure every component is something we’d want contributed? Specifically Locale change banner?_

Let me know what you think about this one, @shixiedesign and I can leave it out if necessary.

janchild commented:
I've pushed up two new versions of this section with your descriptions, @oliviaflory. Here are images of the two options to look at in the meantime.  

I've edited the descriptions down a bit, fyi. ;)  

Of the two, I prefer the table. The information is more compact and having the links beside the description makes it easier to take action. 

![image](https://user-images.githubusercontent.com/44503588/132031183-5e4e11a2-0fc5-404b-8459-032074e24bd5.png)

![image](https://user-images.githubusercontent.com/44503588/132031211-ee72f877-2e48-417d-a404-fa89c51444e0.png)

shixiedesign commented:
Agreed listing the 4 folders out are better than pointing to top level Components folder. While mini card always have the bling because of those cute diamonds, I'd agree that table is clearer here. The definition more connected to the link. 👍

janchild commented:
> Agreed listing the 4 folders out are better than pointing to top level Components folder. While mini card always have the bling because of those cute diamonds, I'd agree that table is clearer here. The definition more connected to the link. 👍

Awesome!! Will get rid of the second option and we'll go with the table. 

@shixiedesign What do you think about Olivia's comment regarding some of the `Design only` components?

janchild commented:
@shixiedesign and @oliviaflory. I'm pretty sure I captured all the feedback before the long weekend. Would you mind doing a (hopefully) final check so I can merge this puppy? 🐕

shixiedesign reviewed (approved):
Looks good! Thank you!
